pkg/apis/autoscaling/annotation_validation_test.go

```diff
@@ -17,95 +17,118 @@ limitations under the License.
 package autoscaling
 
 import (
-	"fmt"
 	"reflect"
 	"testing"
 
-	"knative.dev/pkg/apis"
+	"github.com/google/go-cmp/cmp"
 )
 
 func TestValidateScaleBoundAnnotations(t *testing.T) {
 	cases := []struct {
 		name        string
 		annotations map[string]string
-		expectErr   *apis.FieldError
+		expectErr   string
 	}{{
 		name:        "nil annotations",
 		annotations: nil,
-		expectErr:   nil,
 	}, {
 		name:        "empty annotations",
 		annotations: map[string]string{},
-		expectErr:   nil,
 	}, {
 		name:        "minScale is 0",
 		annotations: map[string]string{MinScaleAnnotationKey: "0"},
-		expectErr:   nil,
 	}, {
 		name:        "maxScale is 0",
 		annotations: map[string]string{MaxScaleAnnotationKey: "0"},
-		expectErr:   nil,
 	}, {
 		name:        "minScale is -1",
 		annotations: map[string]string{MinScaleAnnotationKey: "-1"},
-		expectErr: &apis.FieldError{
-			Message: fmt.Sprintf("Invalid %s annotation value: must be an integer equal or greater than 0", MinScaleAnnotationKey),
-			Paths:   []string{MinScaleAnnotationKey},
-		},
+		expectErr:   "expected 1 <= -1 <= 2147483647: autoscaling.knative.dev/minScale",
 	}, {
 		name:        "maxScale is -1",
 		annotations: map[string]string{MaxScaleAnnotationKey: "-1"},
-		expectErr: &apis.FieldError{
-			Message: fmt.Sprintf("Invalid %s annotation value: must be an integer equal or greater than 0", MaxScaleAnnotationKey),
-			Paths:   []string{MaxScaleAnnotationKey},
-		},
+		expectErr:   "expected 1 <= -1 <= 2147483647: autoscaling.knative.dev/maxScale",
 	}, {
 		name:        "minScale is foo",
 		annotations: map[string]string{MinScaleAnnotationKey: "foo"},
-		expectErr: &apis.FieldError{
-			Message: fmt.Sprintf("Invalid %s annotation value: must be an integer equal or greater than 0", MinScaleAnnotationKey),
-			Paths:   []string{MinScaleAnnotationKey},
-		},
+		expectErr:   "expected 1 <= foo <= 2147483647: autoscaling.knative.dev/minScale",
 	}, {
 		name:        "maxScale is bar",
 		annotations: map[string]string{MaxScaleAnnotationKey: "bar"},
-		expectErr: &apis.FieldError{
-			Message: fmt.Sprintf("Invalid %s annotation value: must be an integer equal or greater than 0", MaxScaleAnnotationKey),
-			Paths:   []string{MaxScaleAnnotationKey},
-		},
+		expectErr:   "expected 1 <= bar <= 2147483647: autoscaling.knative.dev/maxScale",
+	}, {
+		name:        "max/minScale is bar",
+		annotations: map[string]string{MaxScaleAnnotationKey: "bar", MinScaleAnnotationKey: "bar"},
+		expectErr:   "expected 1 <= bar <= 2147483647: autoscaling.knative.dev/maxScale, autoscaling.knative.dev/minScale",
```

Comment:
This string formatting seems strange -- is this the actual error we get?

Reply:
Yep, but this is internal method invoked with "ViaField" which makes output more interesting (i.e. `metadata.annotations` prefix will be added.